src/lib/idmap/sss_idmap.h

```diff
@@ -956,6 +961,136 @@ enum idmap_error_code sss_idmap_smb_sid_to_bin_sid(struct sss_idmap_ctx *ctx,
                                                    struct dom_sid *smb_sid,
                                                    uint8_t **bin_sid,
                                                    size_t *length);
+
+/**
+ * Typedef for functions to calculate an offset for id-mapping and, if
+ * possible, for the reverse operation.
+ */
+typedef enum idmap_error_code (idmap_offset_func)(void *pvt,
+                                                  uint32_t range_size,
+                                                  const char *input,
+                                                  long long *offset);
+
+typedef enum idmap_error_code (idmap_rev_offset_func)(struct sss_idmap_ctx *ctx,
+                                                      void *pvt,
+                                                      uint32_t offset,
+                                                      char **out);
+
+/**
+ * @brief Add a generic domain to the idmap context
+ *
+ * @param[in] ctx         Idmap context
+ * @param[in] domain_name Zero-terminated string with the domain name
+ * @param[in] domain_id   Zero-terminated string representation of a unique
+ *                        identifier of the domain, e.g. if available a domain
+ *                        UUID or the URI of domain specific service
+ * @param[in] range       Id range struct with smallest and largest POSIX id of
+ *                        the range
+ * @param[in] range_id    A name for the id range, currently not used, might
+ *                        become important when we allow multiple ranges for a
+ *                        single domain
+ * @param[in] offset_func Function to calculate an offset in a given range
+ *                        from some input given as string, if NULL
+ *                        offset_murmurhash3 will be used.
+ * @param[in] rev_offset_func Function to calculate the original input from a
+ *                        given offset, i.e. the reverse of offset_func, may
+ *                        be NULL
+ * @param[in] offset_func_pvt Private data for offset_func and
+ *                        rev_offset_func, may be NULL
```

Comment:
Can it say that 'if NULL then sss_idmap_offset_murmurhash3 is used by default'?

Reply:
If 'external_mapping == false'

Reply:
fixed